src/datepicker/datepicker.js

```diff
@@ -118,6 +120,24 @@ export default class Datepicker extends React.Component<
 
   handleInputChange = (event: SyntheticInputEvent<HTMLInputElement>) => {
     const inputValue = event.currentTarget.value;
+    const inputHasNumbers = new RegExp('[1-9]').test(inputValue);
+    const inputHasLetters = new RegExp('[a-z]', 'i').test(inputValue);
+
+    // without letters and numbers, chances are, the user cleared the input
+    // and only the mask is present
```

Comment:
Could we pull out the mask below into a class method and check against that?
```js
  getMask = () => {
    const {formatString} = this.props;
    let mask = null;
    if (this.props.mask !== null) {
      mask =
        // using the mask provided through the top-level API
        this.props.mask ||
        // to make sure it's not a breaking change, we try calculating the input mask
        // from the formatString, if used by the developer

        // 1. mask generation from the formatstring if it's a range input
        (formatString && this.props.range
          ? `${formatString} – ${formatString}`.replace(/[a-z]/gi, '9')
          : null) ||
        // 2. mask generation from the formatstring if it is NOT a range input
        (formatString ? formatString.replace(/[a-z]/gi, '9') : null) ||
        // falling back to the default masks
        (this.props.range ? '9999/99/99 – 9999/99/99' : '9999/99/99');
    }
    return mask;
  };
```
and here 
```js
    // user cleared the input and only the mask is present
    if (inputValue === this.getMask()) {
      if (this.props.onChange) {
        this.props.range
          ? this.props.onChange({
              date: [],
            })
          : this.props.onChange({
              date: null,
            });
      }
      return;
    }
```

Reply:
yeah, that's what I wanted initially - but it's a bit more tricky

the `getMask` function would return something like this: `9999/99/99 – 9999/99/99`, but the inputvalue for this mask will be: `    /  /   –     /  /  `

this is why I ended up with the current logic - what do you think?

Reply:
it may be possible to replace all instances of `9` with ` ` and compare like that, but i'm not sure the extent of change required. I'm comfortable seeing what edge cases shake out based on the implementation in this pr

Reply:
Ohh that’s right. I still think we should check against the mask for exactness, though, in case we have a mask with text like `Pickup: 9999-99-99`. Could we swap out the 9s?
```js
if (inputValue === this.getMask().replace(/9/g, ' ')) {
```

Reply:
that could work - let me try that

Reply:
@leyanlo you were right, this works perfectly - thank you

Reply:
🎉